crates/extensions/c8y_http_proxy/src/tests.rs

```diff
@@ -63,6 +64,256 @@ async fn c8y_http_proxy_requests_the_device_internal_id_on_start() {
     .await;
 }
 
+#[tokio::test]
+async fn retry_internal_id_on_expired_jwt() {
+    let c8y_host = "c8y.tenant.io";
+    let device_id = "device-001";
+    let token = "JWT token";
+    let external_id = "external-device-001";
+    let tmp_dir = "/tmp";
+
+    let (mut proxy, mut c8y) =
+        spawn_c8y_http_proxy(c8y_host.into(), device_id.into(), tmp_dir.into(), token).await;
+
+    // Even before any request is sent to the c8y_proxy
+    // the proxy requests over HTTP the internal device id.
+    let init_request = HttpRequestBuilder::get(format!(
+        "https://{c8y_host}/identity/externalIds/c8y_Serial/{device_id}"
+    ))
+    .bearer_auth(token)
+    .build()
+    .unwrap();
+    c8y.assert_recv(Some(init_request)).await;
+
+    // Cumulocity returns unauthorized error (401), because the jwt token has expired
+    let c8y_response = HttpResponseBuilder::new()
+        .status(401)
+        .json(&InternalIdResponse::new(device_id, external_id))
+        .build()
```

Comment:
```suggestion
        .status(401)
        .build()
```
An unauthorized response wouldn't have this `json` body, right?

Reply:
right, removed